src/py/flwr/server/strategy/qfedavg.py

```diff
@@ -191,8 +191,29 @@ def norm_grad(grad_list: NDArrays) -> float:
         eval_result = self.evaluate(
             server_round, ndarrays_to_parameters(weights_before)
         )
+        # Ensure `loss` is always defined
+        # Aggregate fit results
+        losses = [
+            fit_res.loss
+            for _, fit_res in results
+            if hasattr(fit_res, "loss") and not np.isnan(fit_res.loss)
+        ]
```

Comment:
[FitRes](https://github.com/adap/flower/blob/fe7ac61b5d28fe739a2145c6992a8efe3f500358/src/py/flwr/common/typing.py#L134) objects don't have a `loss` attribute. Therefore `losses` will always be an empty list.

Reply:
removed this elif based on comments.